adapters/lora-vertx/src/main/java/org/eclipse/hono/adapter/lora/providers/FireflyProvider.java

```diff
@@ -0,0 +1,93 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.adapter.lora.providers;
+
+import io.vertx.core.json.JsonObject;
+import org.eclipse.hono.adapter.lora.LoraConstants;
+import org.eclipse.hono.adapter.lora.LoraMessageType;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A LoRaWAN provider with API for Firefly.
+ */
+@Component
+public class FireflyProvider implements LoraProvider {
+
+    private static final String FIELD_FIREFLY_SPREADING_FACTOR = "spreading_factor";
+    private static final String FIELD_FIREFLY_FPORT = "port";
```

Comment:
Did you mean `FIELD_FIREFLY_PORT`?

Reply:
No, this is the "function_port". Renamed it to `FIELD_FIREFLY_FUNCTION_PORT` to be declarative